### PR TITLE
Adds station traits: Small modifiers that can randomly be chosen each round

### PR DESCRIPTION
**Pull request by swissloaf**
https://github.com/tgstation/tgstation/pull/56309
https://github.com/tgstation/tgstation/pull/56718
https://github.com/tgstation/tgstation/pull/56726
https://github.com/tgstation/tgstation/pull/57224
https://github.com/tgstation/tgstation/pull/57532
https://github.com/tgstation/tgstation/pull/57679

# General Documentation

### Intent of your Pull Request

This PR adds station traits which are essentially small changes to a normal round that can be used to create small variations in how a round might play out, sometimes there might be only one, very simple trait, but you might have a round where they have a somewhat bigger impact, to make rounds feel slightly more different from each other.

The PR is mostly a framework atm to probe people to post interesting ideas for station traits, but is already completely functional with a few examples of what kind of things are possible.

The following traits have been added:

Positive:

- Lucky winner - Free pizza and beer every 6-12 minutes
- Galactic Grant - Larger starting funds for cargo
- Premium internals boxes - emergency box has flare + radio as bonus
- Bountiful bounties - Bounties pay 20% more
- Strong supply lines - Imports 20% cheaper
- Filled maint - More loot in maint
- Fast shuttle - Cargo shuttle is faster
- Free scarves - Free scarfs if a slot is free

Neutral:

- Bananium shipment - Clown starts with 5 sheets of bananium (Neutral because this helps noone but the clown)
- Ian adventure - Ian teleports to a random spot on the station
- Glitched PDAs - PDA's have a different beep
- Announcer intern - Changes the announcement messages to sound like they're from an intern at Centcom
- Medbot announcer - Changes the announcements messages to sound like they're from the Medbot

Negative:
- Carp infestation - Carp event is more common and can start earlier
- Weak supply lines - Imports 20% more expensive
- Blackout - Station lights are partially broken around the station
- Empty maint - Less loot in maint
- Overflow bureacracy mistake - Random overflow job (From a vetted list)
- Random spawns - Random spawn location (by drop pod)
- Slow shuttle - Cargo shuttle is slower

image

### Why is this change good for the game?

This PR is good because it adds a little bit of variety to every round! Who knows what you are going to get, be it an intern announcer, a medbot announcer, carp infestation, unique AIs, a fast shuttle, free scarves, who knows!

It's also really cool.

# Wiki Documentation

### Briefly describe your PR and the impacts of it, in layman's terms. 
I did all of this in the "Intent of your PR" section

### What should players be aware of when it comes to the changes your PR is implementing?
Each round will be a little bit different. For instance, if you're a loot hunter in maint, some rounds you may find nothing or find a shit ton more than on a regular round. Nothing is certain and it could have a negative, positive, or neutral effect on your round!

### What general grouping does this PR fall under? 
Changes to the round, changes to the announcement system, changes to cargo, changes to the shuttles, etc..

### If there are any numerical values involved in your PR that will be relevant to a player, please note them here. 
Weights for the events:

Positive traits:
- Lucky winner - weight of 1
- Galactic grant - weight of 5
- Premium internals boxes - Weight of 10
- Bountiful bounties - weight of 5
- Strong supply lines - Weight of 5
- Scarves - Weight of 5
- Filled maint - Weight of 5
- Quick Shuttle - weight of 5

Neutral traits:
- Bananium Shipment - weight of 5
- Ian's Adventure - weight of 5
- Glitched PDA's - weight of 15
- Announcement intern - weight of 1
- Medbot announcer - weight of 1

Negative traits:
- Carp infestation - weight of 5
- Distant supply lines - Weight of 3
- Random spawns - Weight of 2
- Blackout - weight of 3
- Empty maint - weight of 5
- Overflow bureacracy - Weight of 5
- Slow shuttle - weight of 5
- Bot language error - weight of 3

# Changelog

Edit this changelog for changes that are noticeable by the players. Remove it if this isn't the case. If you add a name after the ':cl', that name will be used in the changelog. Leave it empty to use your GitHub name. Prefix the PR title with [admin] if it's something admin related. Prefix the PR title with [s] if you are fixing an exploit so it is not announced on discord and the server.

:cl:  
rscadd: Adds station traits
soundadd: Adds two new announcer sounds: Intern and medbot
/:cl:

### CODE REVIEW

**Review by redmoogle — APPROVED** [review]
Code looks fine, undergoing test w/ swiss

**Review by Partheo — APPROVED** [review]
Consil or lorepeople might want to take a look at them, specifically the intern AI sound effects. Funny, but maybe not what we want. Personally I like them.

**Comment by Ratvar** [comment]
-Unique AI - Random lawset at roundstart for AI- already exists, kinda

File: code/datums/station_traits/positive_traits.dm [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	show_in_report = TRUE
+	report_message = "The internals boxes for your crew have been filled with bonus equipment."
+	trait_to_give = STATION_TRAIT_PREMIUM_INTERNALS
+
+/datum/station_trait/bountiful_bounties
+	name = "Bountiful bounties"
+	trait_type = STATION_TRAIT_POSITIVE
+	weight = 5
+	show_in_report = TRUE
+	report_message = "It seems collectors in this system are extra keen to on bounties, and will pay more to see their completion."
```
Comment (redmoogle) [inline code/datums/station_traits/positive_traits.dm]:
```suggestion
	report_message = "It seems collectors in this system are extra keen to providing bounties, and will pay more to see their completion."
```
Something like this, previous sentence didn't make sense

File: code/__HELPERS/priority_announce.dm [hunk 2]
```diff
@@ -27,6 +29,11 @@
 				GLOB.news_network.SubmitArticle(title + "<br><br>" + text, "Central Command", "Station Announcements", null)
 
 	announcement += "<br><span class='alert'>[html_encode(text)]</span><br>"
```
Comment (redmoogle) [inline code/__HELPERS/priority_announce.dm]:
```suggestion
```
found the broken line

**Review by redmoogle — CHANGES_REQUESTED** [review]

File: code/__HELPERS/priority_announce.dm (hunk 2 above)
Comment (redmoogle) [reply code/__HELPERS/priority_announce.dm]: fully omit it, it's done 2 lines below, hence double announcements

Comment (swissloaf) [reply code/__HELPERS/priority_announce.dm]: Damn byodn code!!

**Comment by swissloaf** [comment]
> -Unique AI - Random lawset at roundstart for AI- already exists, kinda

Handled, forgot we had this shit. It was a couple lines down anyway, although maybe in the future it might be good to have it under the station traits system

**Comment by athena413** [comment]
I took a look at the Intern sound files and i _really_ don't like them

**Review by redmoogle — APPROVED** [review]

**Comment by SomeguyManperson** [comment]
fix conflict

**Comment by N3D6** [comment]
I like this but the medibot announcer is geniunely awful

**Comment by swissloaf** [comment]
Weird thingy runtime with announcer intern

**Comment by JamieD1** [comment]
Remove Late arrivals, causes lag fest.

**Review by JamieD1 — CHANGES_REQUESTED** [review]
Remove Late Arrivals

**Comment by Cartlord** [comment]
Appears that the premium internals box modifier doesn't do anything to help the boxes of Shaft miners, likely because theirs have non-standard contents.

**Comment by swissloaf** [comment]
> Remove Late Arrivals

donezo

**Comment by alao3alao3** [comment]
https://github.com/yogstation13/Yogstation/blob/8fa0e59904e7ada3149b14d6b4a20229d23fcd5a/code/datums/station_traits/negative_traits.dm
dpng
ain't got prisoner

**Comment by JamieD1** [comment]
Theos: It should work I think

**Comment by Cartlord** [comment]
Premium internals can trigger twice at once, cancelling itself out

**Comment by redmoogle** [comment]
> Premium internals can trigger twice at once, cancelling itself out

Make an issue dont necro

**Pull request merged**